ipa-core/src/cli/verbosity.rs

```diff
@@ -32,24 +32,29 @@ impl Verbosity {
     #[must_use]
     pub fn setup_logging(&self) -> LoggingHandle {
         let filter_layer = self.log_filter();
+        info!("Logging setup at level {}", filter_layer);
+
         let fmt_layer = fmt::layer()
             .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
             .with_ansi(std::io::stderr().is_terminal())
             .with_writer(stderr);
 
-        tracing_subscriber::registry()
-            .with(self.log_filter())
-            .with(fmt_layer)
-            .with(MetricsLayer::new())
-            .init();
+        let registry = tracing_subscriber::registry()
+            .with(filter_layer)
+            .with(fmt_layer);
+
+        if cfg!(feature = "disable-metrics") {
```

Comment:
nit: I feel it would be better to have the `init()` outside of the if-else. Mostly for future features that might modify the struct.

```
if cfg!(not(feature = "disable-metrics")) {
registry.with(MetricsLayer::new())
}
registry.init();
```